### ISSUE TITLE
Rule Proposal: vue/no-undefined

### ISSUE THREAD

**Issue by golopot** [vuejs/eslint-plugin-vue#875]
**Please describe what the rule should do:**

Forbid usages of undefined variable in vue template.

**What category should the rule belong to?**

- [ ] Enforces code style
- [x] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<template>
  <button @click="handleClick">foo</button>
  <!--            ~~~~~~~~~~~           -->
  <!--            undefined variable    -->
</template>

<script>
export default {}
</script>
```

**Comment by g-plane** [comment]
It's hard to detect, because Vue allows you to use mixins which can't be statically analyzed easily.
